### DISCUSSION TITLE
NewFeature: Cross-launch into maps

### DISCUSSION THREAD

**Discussion by kingsglade** [oguzhnatly/flutter_carplay#47] [Q&A] [answered]
Hi!

Firstly thanks for the excellent package, I'm having good success adding CarPlay support to my existing train times app [https://www.nextrain.co.uk](url).  

Apologies if this has been asked before but one of my requirements is to cross-launch into Maps / Navigation.  I have coordinates for all UK stations so it should be just a matter of following similar to this strategy that supports google/Apple Maps [https://developer.apple.com/forums/thread/702484](url).

Ideally this would be something I could invoke directly so I can hang it off of an action sheet but alternatively/also it could be a "Go" button on the POI template.

Any thoughts on whether & when this can be supported?

Many thanks
Mark

**Comment by oguzhnatly** [comment, marked as answer]
For cross launching into Apple Maps or other navigation apps from CarPlay, you can use URL schemes from your Flutter side:

```dart
// Launch Apple Maps with coordinates
final url = 'http://maps.apple.com/?daddr=${lat},${lng}&dirflg=d';
launchUrl(Uri.parse(url));
```

This works from within your CarPlay app. The navigation will open in the Maps app on the CarPlay display. This is an app level implementation rather than a plugin feature, since it's just URL scheme launching.
